lms/djangoapps/shoppingcart/models.py

```diff
@@ -824,6 +824,69 @@ def generate_pdf_invoice(self, course, course_price, quantity, sale_price):
 
         return pdf_buffer
 
+    def __unicode__(self):
+        return "company: {} , internal reference: {} , customer reference number: {}".format(
+            self.company_name,
+            self.internal_reference,
+            self.customer_reference_number
+        )
+
+INVOICE_TRANSACTION_STATUSES = (
+
+    ('started', 'started'),
+    ('completed', 'completed'),
+    ('cancelled', 'cancelled'),
+    ('payment', 'payment'),
+    ('refund', 'refund'),
```

Comment:
I'm not sure I'm convinced that using these states will make transactions easier to query.  First, the cardinality of this table is likely to stay relatively low (hundreds, maybe thousands of entries), so I don't know if we need to be too worried about indexing the status.  Second, one of the most important queries we're going to want to run on these models is "what is the current balance?", which is easier if we can just sum up all the transactions (positive and negative) for an invoice.

A bigger concern here is that "started", "completed", and "cancelled" are NOT the same kind of thing as "payment" / "refund".  It seems to me that if we're going to maintain both kinds of states, they at least need to be stored in separate fields.

If we worked strictly with amounts, then the statuses would mean:

* Positive amount, "started": Someone said they're sending us a check, but we haven't received it yet.
* Positive amount, "completed": We received and deposited a check.
* Positive amount, "cancelled": Someone said they're sending us a check, but it never arrived / it bounced / it was made to the wrong person, etc.
* Negative amount, "started": We sent someone a refund.
* Negative amount, "completed": We sent someone a refund and they deposited it.
* Negative amount, "cancelled": We sent someone a refund but voided the check before it was deposited.